File: frontend/src/types/climate.ts
/** Climate risk assessment for a township. */
export interface ClimateRiskResponse {
  township_id: string;
  township_name: string;
  season: string;
  drought_probability: number;
  flood_probability: number;
  temp_anomaly_celsius: number;
  rainfall_forecast_mm: number;
  rainfall_historical_avg_mm: number;
  risk_level: string;
  data_quality_score: number;
  data_source: string;
}

File: backend/app/api/v1/schemas/climate.py
"""Pydantic schemas for climate risk API."""

from __future__ import annotations

from pydantic import BaseModel


class ClimateRiskResponse(BaseModel):
    """Climate risk assessment response for a township."""

    township_id: str
    township_name: str
    season: str
    drought_probability: float
    flood_probability: float
    temp_anomaly_celsius: float
    rainfall_forecast_mm: float
    rainfall_historical_avg_mm: float
    risk_level: str
    confidence: float
    data_source: str
